org.freedesktop.Sdk.Extension.openjfx.yaml

```diff
@@ -0,0 +1,43 @@
+id: org.freedesktop.Sdk.Extension.openjfx
+branch: '23.08'
+runtime: org.freedesktop.Sdk
+runtime-version: '23.08'
+build-extension: true
+sdk: org.freedesktop.Sdk
+separate-locales: false
+appstream-compose: false
+build-options:
+  no-debuginfo: true
+  prefix: /usr/lib/sdk/openjfx
+
+modules:
+  - name: javafx_jmods
+    buildsystem: simple
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://download.bell-sw.com/java/21.0.2+14/bellsoft-jdk21.0.2+14-linux-amd64-full.tar.gz
+        dest-filename: java-openjfx.tar.gz
+        sha512: 1b349703a585198af2ae56e2ec4ba1922c794be444bdf8abc6252a64a8cfe5bf26714971668ef8481f8223a3ff2a4bb14a1ba0cfe965eea1c32791e8cdb205f1
+      - type: file
+        only-arches:
+          - aarch64
+        url: https://download.bell-sw.com/java/21.0.2+14/bellsoft-jdk21.0.2+14-linux-aarch64-full.tar.gz
+        dest-filename: java-openjfx.tar.gz
+        sha512: 043e798e02433c03779a95c9583881908a5a5e12c9c2200fd18da96a24e48103d503795b4a6c191b5cf74f4fcda2e6b04c25336799c15d59f61e6832b73bf400
+    build-commands:
+#      - tar xfv java-openjfx.tar.gz --wildcards "jdk-21.0.2-full/jmods/javafx.*.jmod" -C $FLATPAK_DEST --strip-components 1
+      - tar xfv java-openjfx.tar.gz --wildcards "jdk-21.0.2-full/jmods/javafx.*.jmod" --strip-components 1
+      - mv jmods $FLATPAK_DEST
+
```

Comment:
I think you need some scripts like this to let apps using the extension, easily copy everything to app and set, $PATH etc. https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk21/blob/30eb0497b1f21d250b7ef63f7b829838baba14e3/org.freedesktop.Sdk.Extension.openjdk21.yaml#L172-L195

Reply:
jmods are not meant to be used at run time, in fact they can not. So there is nothing to copy.

They are used at compile and link time. Like in this sample here:
https://github.com/trixon/flatpak-java-cri/blob/1bcfb63706da7e1ff8bc5abc814085765454e8ce/se.trixon.HelloFx.yml#L19C1-L29C26

The runtime for that demo is 60MB, compare that to the copying of jre+javafx that lands on about 280MB.

The recommended way to distribute java applications today is to provide a custom runtime image, not the full blown "jre".

Reply:
notice how this is `Sdk` and not runtime? It's for **building**.

Also the prefered way on flathub is to ship the JDK that come with the corresponding SDK Extension. That allow supporting more than just x86_64.

Reply:
And also allow easier security updates.

Reply:
> notice how this is Sdk and not runtime? It's for building.

SDK extensions are not only for building though, all the regular JDK extensions use it and still give the option of copying the contents because they are required at runtime too.

Node extensions also use SDK but can be both used at build and runtime (for the `node` executable)

Anyway if this doesn't require it, it's fine.